#! /bin/sh /usr/share/dpatch/dpatch-run
## 24_fix_64bit_times.dpatch by  <robin@debian>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix time display on 64bit builds

@DPATCH@

diff -U3 -r linden.new/indra/llcommon/lltimer.cpp linden/indra/llcommon/lltimer.cpp
--- linden.new/indra/llcommon/lltimer.cpp	2007-10-20 00:45:55.000000000 +0100
+++ linden/indra/llcommon/lltimer.cpp	2007-10-21 10:41:17.000000000 +0100
@@ -445,20 +445,20 @@
 {
 	time_t unix_time = (time_t)utc_time;
 
-	S32 pacific_offset_hours;
+	U32 pacific_offset_hours;
 	if (pacific_daylight_time)
 	{
-		pacific_offset_hours = -7;
+		pacific_offset_hours = 7;
 	}
 	else
 	{
-		pacific_offset_hours = -8;
+		pacific_offset_hours = 8;
 	}
 
 	// We subtract off the PST/PDT offset _before_ getting
 	// "UTC" time, because this will handle wrapping around
 	// for 5 AM UTC -> 10 PM PDT of the previous day.
-	unix_time += pacific_offset_hours * MIN_PER_HOUR * SEC_PER_MIN;
+	unix_time -= pacific_offset_hours * MIN_PER_HOUR * SEC_PER_MIN;
  
 	// Internal buffer to PST/PDT (see above)
 	struct tm* internal_time = gmtime(&unix_time);
